pkg/docker/docker.go

```diff
@@ -665,39 +789,90 @@ func determineCommandBaseDir(opts RunContainerOptions, imageMetadata *docker.Ima
 }
 
 // dumpContainerInfo dumps information about a running container (port/IP/etc).
-func dumpContainerInfo(container *docker.Container, d *stiDocker, image string) {
-	cont, icerr := d.client.InspectContainer(container.ID)
+func dumpContainerInfo(container *dockertypes.ContainerCreateResponse, d *stiDocker, image string) {
+	containerJSON, err := d.kubeDockerClient.InspectContainer(container.ID)
+	if err != nil {
+		return
+	}
+
 	liveports := "\n\nPort Bindings:  "
-	if icerr == nil {
-		// Ports is of the following type:  map[docker.Port][]docker.PortBinding
-		for port, bindings := range cont.NetworkSettings.Ports {
-			liveports = liveports + "\n  Container Port:  " + port.Port()
-			liveports = liveports + "\n        Protocol:  " + port.Proto()
-			liveports = liveports + "\n        Public Host / Port Mappings:"
-			for _, binding := range bindings {
-				liveports = liveports + "\n            IP: " + binding.HostIP + " Port: " + binding.HostPort
-			}
+	for port, bindings := range containerJSON.NetworkSettings.NetworkSettingsBase.Ports {
+		liveports = liveports + "\n  Container Port:  " + string(port)
+		for _, binding := range bindings {
+			liveports = liveports + "\n            IP: " + binding.HostIP + " Port: " + binding.HostPort
 		}
-		liveports = liveports + "\n"
 	}
+	liveports = liveports + "\n"
 	glog.V(0).Infof("\n\n\n\n\nThe image %s has been started in container %s as a result of the --run=true option.  The container's stdout/stderr will be redirected to this command's glog output to help you validate its behavior.  You can also inspect the container with docker commands if you like.  If the container is set up to stay running, you will have to Ctrl-C to exit this command, which should also stop the container %s.  This particular invocation attempts to run with the port mappings %+v \n\n\n\n\n", image, container.ID, container.ID, liveports)
 }
 
+// b copy from kube_docker_client.go ... essentially, if that code could give is a way to inspect the error from engin-api's ContainerAttach call prior
+// to blocking on the IO, we could just leverage the kube_docker_client.go code entirely ... essentially a "non-blocking" attach like go-dockerclient provided
```

Comment:
or is // b supposed to mean // begin ?  (figured that out based on the // e at the end, i think it could be made more obvious)